clarify how to get binaries if F5 results in error

hirehamir opened the pull request.

hirehamir committed:
clarify how to get binaries if F5 results in error

sean-mcmanus commented on Documentation/Building the Extension.md:
```diff
@@ -13,3 +13,12 @@ Prerequisite steps:
   * Open the **Extension** folder in Visual Studio Code and press F5. This will launch a VS Code Extension Host window and activate the TypeScript debugger. You can set breakpoints on the extension source code and debug your scenario.
+      * If, after pressing F5, you see the following error in the `[Extension Development Host]` window,
+      > Unable to start the C/C++ language server. IntelliSense features will be disabled. Error: Missing binary at .../vscode-cpptools/Extension/bin/cpptools
+      * ... Then, you can follow the instructions in this [comment in a discussion about building this extension locally](https://github.com/microsoft/vscode-cpptools/discussions/8745#discussioncomment-2091563).
```
Can you remove this "..." (not sure why it's there)?

hirehamir replied:
Ah, I just added those to signify that `/vscode-cpptools/Extension/bin/cpptools` could be part of a much longer path, and didn't necessarily start at the root directory.

[05400ba](https://github.com/microsoft/vscode-cpptools/pull/8788/commits/05400ba8a2ac80dfb7e3b4232edc3c3d08b01e7d) addresses this.

sean-mcmanus replied:
No, you misunderstood. I mean the "..." before "Then", not the "..." in the path, which was fine.

hirehamir replied:
Ah, I understand. Thank you for the clarification.

86df44e should address this.

sean-mcmanus commented on Documentation/Building the Extension.md:
```diff
@@ -13,3 +13,12 @@ Prerequisite steps:
+      * ... Then, you can follow the instructions in this [comment in a discussion about building this extension locally](https://github.com/microsoft/vscode-cpptools/discussions/8745#discussioncomment-2091563).
+        > get the 1.8.2 binaries from installing the extension and then copying the binaries
+        1. To do this, install this extension from the Visual Studio Marketplace and find its location on your device. It might be in a directory like `\\wsl$\Ubuntu\home\hamir\.vscode-server\extensions\ms-vscode.cpptools-1.8.2`, for example.
+        2. Next, go to the `bin/` directory of the aforementioned directory, and drag-and-drop, or copy-and-paste, `cpptools` and `cpptools-srv` from `...\extensions\ms-vscode.cpptools-1.8.2\bin\` to this repository's `Extension\bin\` directory on your local device, so that `.../vscode-cpptools/Extension/bin/cpptools` and `.../vscode-cpptools/Extension/bin/cpptools-srv` both exist in your workspa ...
```
Can you replace 1.8.2 with something like `<package.version.number>`?

hirehamir replied:
Sure! [3d509a2](https://github.com/microsoft/vscode-cpptools/pull/8788/commits/3d509a2d18340583589cdec10266f9971edd4e0f) should address this.

sean-mcmanus commented on Documentation/Building the Extension.md:
```diff
@@ -13,3 +13,12 @@ Prerequisite steps:
+        1. To do this, install this extension from the Visual Studio Marketplace and find its location on your device. It might be in a directory like `\\wsl$\Ubuntu\home\hamir\.vscode-server\extensions\ms-vscode.cpptools-1.8.2`, for example.
+        2. Next, go to the `bin/` directory of the aforementioned directory, and drag-and-drop, or copy-and-paste, `cpptools` and `cpptools-srv` from `...\extensions\ms-vscode.cpptools-1.8.2\bin\` to this repository's `Extension\bin\` directory on your local device, so that `.../vscode-cpptools/Extension/bin/cpptools` and `.../vscode-cpptools/Extension/bin/cpptools-srv` both exist in your workspa ...
+        3. The aforementioned warning should be gone, and Intellisense, which gives those squiggly red error lines, should now be present.
+
```
Can you add a 4th point that says something like "The "insiders" branch has binaries compatible with the latest Pre-Release version of the extension, and the "release" branch has binaries compatible with the latest Release version, but the "main" branch may have TypeScript changes that are incompatible with the published binaries, in which case, you'll need to create a branch off the "insiders" or "release" branches."?

hirehamir replied:
Absolutely! [d4c4a10](https://github.com/microsoft/vscode-cpptools/pull/8788/commits/d4c4a102f7762a34700e6f93e5c529de05e28e0a) should address this.

sean-mcmanus reviewed (changes requested):
Thanks for helping improve the docs.

sean-mcmanus reviewed (changes requested):
Also, we should add a line about possibly needing to copy over the debugAdapters folder (i.e. cl.exe debugging wasn't working for me until after I did that). @WardenGnaw Any other debugger related steps should be documented?

WardenGnaw commented on Documentation/Building the Extension.md:
```diff
@@ -13,3 +13,12 @@ Prerequisite steps:
   * **(optional)** Set an environment variable `CPPTOOLS_DEV=1`.
```
Lines 13 14 can be removed due to https://github.com/microsoft/vscode-cpptools/commit/62fff2b37b0ac5d3f04e01192191731db1c732b6#diff-c593d2f539ed1508705a877abb8a2d2b935d53950f8bde51ad05ac84bc6f2b18

hirehamir replied:
Thanks for the suggestion! [888efa9](https://github.com/microsoft/vscode-cpptools/pull/8788/commits/888efa96850d9719c7f58e30fb2d3b88cd3d9314) removed lines 13 and 14.

hirehamir committed:
Merge branch 'main' into patch-1

hirehamir committed:
remove `...` from a couple paths

This commit addresses https://github.com/microsoft/vscode-cpptools/pull/8788#discussion_r800937204.

hirehamir committed:
replace `1.8.2` with `<package.version.number>`

This commit addresses https://github.com/microsoft/vscode-cpptools/pull/8788#discussion_r800938662.

hirehamir committed:
add a recommended fourth point

This commit addresses https://github.com/microsoft/vscode-cpptools/pull/8788#discussion_r800942615.

hirehamir committed:
remove lines that are no longer needed

This commit addresses https://github.com/microsoft/vscode-cpptools/pull/8788#discussion_r804083255.

sean-mcmanus commented on Documentation/Building the Extension.md:
```diff
@@ -10,6 +10,14 @@ Prerequisite steps:
+        1. To do this, install this extension from the Visual Studio Marketplace and find its location on your device. It might be in a directory like `\\wsl$\Ubuntu\home\hamir\.vscode-server\extensions\ms-vscode.cpptools-<package.version.number>`, for example.
+        2. Next, go to the `bin/` directory of the aforementioned directory, and drag-and-drop, or copy-and-paste, `cpptools` and `cpptools-srv` from `\extensions\ms-vscode.cpptools-<package.version.number>\bin\` to this repository's `Extension\bin\` directory on your local device, so that `/vscode-cpptools/Extension/bin/cpptools` and `/vscode-cpptools/Extension/bin/cpptools-srv` both exist in yo ...
+        3. The aforementioned warning should be gone, and Intellisense, which gives those squiggly red error lines, should now be present.
+        4. The `insiders` branch has binaries compatible with the latest Pre-Release version of the extension, and the `release` branch has binaries compatible with the latest Release version, but the `main` branch may have TypeScript changes that are incompatible with the published binaries, in which case, you'll need to create a branch off the `insiders` or `release` branches."
```
The `"` at the end should be removed.

hirehamir replied:
Ah, I missed that! Thank you for catching it.

[27b1ace](https://github.com/microsoft/vscode-cpptools/pull/8788/commits/27b1ace1ad0088e3b79f6b76e44275fcefaf56a5) should address this.

hirehamir committed:
place `...` in appropriate places

This commit addresses https://github.com/microsoft/vscode-cpptools/pull/8788#discussion_r806136131.

hirehamir committed:
fix stray quotation mark `"`

This commit addresses https://github.com/microsoft/vscode-cpptools/pull/8788#discussion_r806134843.

sean-mcmanus committed:
Merge branch 'main' into patch-1

sean-mcmanus committed:
Merge branch 'main' into patch-1
